### ISSUE TITLE
unnecessary_to_owned false positive

### ISSUE THREAD

**Issue by crumblingstatue** [rust-lang/rust-clippy#8759]
### Summary

```Rust
#[derive(Default)]
struct View {
    
}

impl std::borrow::ToOwned for View {
	type Owned = View;
	fn to_owned(&self) -> Self::Owned {
		View {}
	}
}

#[derive(Default)]
struct RenderWindow {
    default_view: View,
}

impl RenderWindow {
    fn default_view(&self) -> &View {
        &self.default_view
    }
    fn set_view(&mut self, _view: &View) {
		
    }
}

fn main() {
	let mut rw = RenderWindow::default();
	rw.set_view(&rw.default_view().to_owned());
}
```

Clippy output:

```
warning: unnecessary use of `to_owned`
  --> src/main.rs:29:14
   |
29 |     rw.set_view(&rw.default_view().to_owned());
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: use: `rw.default_view()`

```
If we apply the suggestion, rustc output:
```
error[E0502]: cannot borrow `rw` as mutable because it is also borrowed as immutable
  --> src/main.rs:29:2
   |
29 |     rw.set_view(rw.default_view());
   |     ^^^--------^-----------------^
   |     |  |        |
   |     |  |        immutable borrow occurs here
   |     |  immutable borrow later used by call
   |     mutable borrow occurs here
   ```

### Version

```text
rustc 1.62.0-nightly (69a5d2481 2022-04-27)
binary: rustc
commit-hash: 69a5d2481e856a5a18885390b8cf6950b9ff8dd3
commit-date: 2022-04-27
host: x86_64-unknown-linux-gnu
release: 1.62.0-nightly
LLVM version: 14.0.1
```
